src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityItemBus.java

```diff
@@ -249,68 +254,49 @@ public void registerAbilities(List<IItemHandlerModifiable> abilityList) {
     }
 
     @Override
-    protected ModularUI createUI(EntityPlayer entityPlayer) {
-        int rowSize = (int) Math.sqrt(getInventorySize());
-        return createUITemplate(entityPlayer, rowSize)
-                .build(getHolder(), entityPlayer);
+    public boolean usesMui2() {
+        return true;
     }
 
-    private ModularUI.Builder createUITemplate(EntityPlayer player, int gridSize) {
-        int backgroundWidth = gridSize > 6 ? 176 + (gridSize - 6) * 18 : 176;
-        int center = backgroundWidth / 2;
-
-        int gridStartX = center - (gridSize * 9);
-
-        int inventoryStartX = center - 9 - 4 * 18;
-        int inventoryStartY = 18 + 18 * gridSize + 12;
-
-        Builder builder = ModularUI.builder(GuiTextures.BACKGROUND, backgroundWidth, 18 + 18 * gridSize + 94)
-                .label(10, 5, getMetaFullName());
-
-        for (int y = 0; y < gridSize; y++) {
-            for (int x = 0; x < gridSize; x++) {
-                int index = y * gridSize + x;
-
-                builder.widget(new SlotWidget(isExportHatch ? exportItems : importItems, index,
-                        gridStartX + x * 18, 18 + y * 18, true, !isExportHatch)
-                                .setBackgroundTexture(GuiTextures.SLOT));
+    @Override
+    public ModularPanel buildUI(GuiCreationContext guiCreationContext, GuiSyncManager guiSyncManager,
+                                boolean isClient) {
+        int rowSize = (int) Math.sqrt(getInventorySize());
+        guiSyncManager.registerSlotGroup("item_inv", rowSize);
+
+        int backgroundWidth = rowSize > 6 ? 176 + (rowSize - 6) * 18 : 176;
+
+        List<List<IWidget>> widgets = new ArrayList<>();
+        for (int i = 0; i < rowSize; i++) {
+            widgets.add(new ArrayList<>());
+            for (int j = 0; j < rowSize; j++) {
+                widgets.get(i)
+                        .add(new ItemSlot()
+                                .slot(SyncHandlers.itemSlot(isExportHatch ? exportItems : importItems, i * rowSize + j)
+                                        .slotGroup("item_inv")));
```

Comment:
there needs to be a filter here for if the slot is for an export hatch, as currently you can insert into export hatches